File: native-apps/ios/Sources/GeregeShellKit/AuthModels.swift
import Foundation

public enum LoginPhase: Equatable, Sendable {
    case idle
    case starting
    case waiting(verificationCode: String, deviceLinkURL: URL?)
    case success
    case expired
    case refused
    case error(String)
}

public struct SessionCookie: Equatable, Sendable {
    public let name: String
    public let value: String
    public let domain: String
    public let path: String
    public let secure: Bool
    public let httpOnly: Bool

    public init(name: String, value: String, domain: String, path: String, secure: Bool, httpOnly: Bool) {
        self.name = name; self.value = value; self.domain = domain; self.path = path
        self.secure = secure; self.httpOnly = httpOnly
    }
}

struct EIDStart: Decodable, Sendable {
    let sessionID: String
    let deviceLinkURL: URL?
    let verificationCode: String
    let expiresAt: Date?

    enum CodingKeys: String, CodingKey {
        case sessionID = "session_id"
        case deviceLinkURL = "device_link_url"
        case verificationCode = "verification_code"
        case expiresAt = "expires_at"
    }
}

struct EIDPoll: Decodable, Sendable { let state: String }

File: native-apps/mobile/android/core/src/main/kotlin/mn/gerege/nexus/core/auth/AuthModels.kt
package mn.gerege.nexus.core.auth

sealed interface LoginPhase {
    data object Idle : LoginPhase
    data object Starting : LoginPhase
    // expiresAtMillis нь UI-гийн countdown-д: сервер хугацаа өгөөгүй бол watch()
    // өөрийн 15 минутын deadline-аа тавьдаг тул энд үргэлж утгатай ирдэг.
    data class Waiting(
        val verificationCode: String,
        val deviceLinkUrl: String?,
        val expiresAtMillis: Long? = null,
    ) : LoginPhase
    data object Success : LoginPhase
    data object Expired : LoginPhase
    data object Refused : LoginPhase
    // message нь хэрэглэгчид харуулах бичвэр; detail нь тэр алдааны техник
    // хэлбэр (exception-ийн бүрэн бичвэр) — зөвхөн «Диагностик» дор гарна.
    data class Error(val message: String, val detail: String? = null) : LoginPhase
}

data class SessionCookie(
    val name: String,
    val value: String,
    val domain: String,
    val path: String,
    val secure: Boolean,
    val httpOnly: Boolean,
)
